scripts/tests/py/metadata.py

```diff
@@ -0,0 +1,223 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+import re
+import yaml
+from typing import Optional
+from typing import Dict
+from typing import List
+from typing import Union
+
+@dataclass
+class Metadata:
+    py_script_path: Optional[str] = None
+    run: Optional[str] = None
+    app: Optional[str] = None
+    factoryreset: Optional[bool] = False
+    comissioning_method: Optional[str] = None
+    discriminator: Optional[int] = None
+    kvs: Optional[str] = None
+    storage_path: Optional[str] = None
+    on_network_commission: Optional[str] = None
+    passcode: Optional[int] = None
+    endpoint: Optional[str] = None
+    manual_code: Optional[str] = None
+    PICS: Optional[str] = None
+    tests: Optional[str] = None
+
+
+class Metadata_Reader:
+    """
+    A class to parse run arguments from the test scripts and 
+    resolve them to environment specific values.
+
+    Attributes:
+    
+    env: str
+       A dictionary that reprsents a environment configuration in
+       YAML format. 
+    """
+    
+    def __init__(self, env_file_name: str):
+        """
+        Constructs the environment object
+
+        Parameters:
+        
+        env_file_name: str
+          Path to the environment file that contains the YAML configuration.
+        """
+        self.env = self.__build_env_object__(env_file_name)
+        
+
+
+    def __build_env_object__(self, env_yaml_file_path: str) -> Dict[str, Union[str, int, Dict]]:
+        """
+        Reads a YAML file and converts it into a dictionary
+
+        Parameters:
+
+        env_yaml_file_path: str
+          path to the YAML file that contains environment configuration
+
+        Returns:
+         Dict - a Python Dictionary representation of the environment YAML file
+        
+        """
+        with open(env_yaml_file_path) as stream:
+            return yaml.safe_load(stream)
+
+    
+    def __resolve_env_vals__(self, metadata_dict: Dict[str, str]) -> None:
+        """
+        Resolves the argument defined in the test script to environment values.
+        For example, if a test script defines "all_clusters" as the value for app
+        name, we will check the environment configuration to see what raw value is
+        assocaited with the "all_cluster" variable and set the value for "app" option
+        to this raw value.
+
+        Parameter:
+        
+        metadata_dict: Dict[str, str]
+          Dictionary where each key represent a particular argument and its value represent
+          the value for that argument defined in the test script.
+
+        Return:
```

Comment:
remove this bit: the `Return: none` uses up 3 lines of text and it is obvious from the signature.

please try to make the code concise so that is easier to read.